packages/babel-plugin-strip-runtime/src/__tests__/jsx-pragma.test.ts

```diff
@@ -0,0 +1,370 @@
+            <div css={{ fontSize: 12, color: 'blue' }}>
+              hello world 2
+            </div>
+          );
+
+          const Component2 = () => (
+            <div css={css({ fontSize: 12, color: 'pink' })}>
+              hello world 2
+            </div>
+          );
+        `;
+
+        expect(() =>
+          transform(code, {
+            run: 'both',
+            runtime: 'classic',
+          })
+        ).toThrow(/Found a `jsx` function call/);
+      });
+    });
+  });
+
+  describe('without JSX pragma', () => {
+    it('throws error if pragma is set in babel config', () => {
+      const code = `
+        /** @jsx jsx */
+        import { css, jsx } from '@compiled/react';
+
+        const Component = () => (
+          <div css={{ fontSize: 12, color: 'blue' }}>
+            hello world 2
+          </div>
+        );
+
+        const Component2 = () => (
+          <div css={css({ fontSize: 12, color: 'pink' })}>
+            hello world 2
+          </div>
+        );
+      `;
+
+      const codeWithRenamedImport = `
+        /** @jsx myJsx */
+        import { css, jsx as myJsx } from '@compiled/react';
+
+        const Component = () => (
+          <div css={{ fontSize: 12, color: 'blue' }}>
+            hello world 2
+          </div>
+        );
+
+        const Component2 = () => (
+          <div css={css({ fontSize: 12, color: 'pink' })}>
+            hello world 2
+          </div>
+        );
+      `;
+
+      for (const c of [code, codeWithRenamedImport]) {
+        expect(() =>
+          transform(c, {
+            run: 'both',
+            runtime: 'classic',
+            babelJSXPragma: 'jsx',
+          })
+        ).toThrow(/Found a `jsx` function call/);
+      }
+    });
+  });
+});
+
+describe('Compiled setup with automatic runtime', () => {
+  describe('with JSX pragma', () => {
+    describe('if Compiled is used in file', () => {
+      it('imports JSX runtime from React, not Compiled', () => {
+        const code = `
+          /** @jsxImportSource @compiled/react */
+          import { css } from '@compiled/react';
+
+          const Component = () => (
+            <div css={{ fontSize: 12, color: 'blue' }}>
+              hello world 2
+            </div>
+          );
+
+          const Component2 = () => (
+            <div css={css({ fontSize: 12, color: 'pink' })}>
+              hello world 2
+            </div>
+          );
+        `;
+
+        const actual = transform(code, {
+          run: 'both',
+          runtime: 'automatic',
+        });
+
+        expect(actual).toContain("import { jsx as _jsx } from 'react/jsx-runtime';");
+        // jsx function calls from React have "PURE" comments
+        // added beforehand from @babel/preset-react. This helps
+        // with tree-shaking by Webpack or Parcel
+        expect(actual).toContain('/*#__PURE__*/ _jsx');
+
+        expect(actual).not.toContain(
+          "import { jsxs as _jsxs } from '@compiled/react/jsx-runtime';"
```

Comment:
We should also check for jsx — perhaps we just check the jsx-runtime module isn't in scope?

```suggestion
          "@compiled/react/jsx-runtime"
```